docs/specs/IDE-execution.md

```diff
@@ -88,6 +89,26 @@ If the execution session is created successfully, the return status code should
 
 If the session cannot be created, appropriate 4xx or 5xx status code should be returned. The response might also return a description of the problem as part of the status line, or in the response body.
 
+### Launch configurations
+
+The run session creation request contains one or more launch configurations for the session. 
+
+The following launch configuration types are recognized by Visual Studio IDE:
+
+**Project launch configuration** <br/>
+
+Project launch configuration contains details for launching programs that have project files compatible with Visual Studio IDE.
+
+| Property | Description | Required? |
+| --- | --------- | --- |
+| `type` | Launch configuration type indicator; must be `project`. | Required |
+| `project_path` | Path to the project file for the program that is being launched. | Required |
+| `mode` | Specifies the launch mode. Currently supported modes are `Debug` (run the project under the debugger) and `NoDebug` (run the project without debugging). | Optional, defaults to `Debug`. |
+| `launch_profile` | Invocation arguments for the program (modeled as array of strings). | Optional |
```

Comment:
As far as I understand, this property is now unused (as well as `disable_launch_profile`). For example, if you start an Aspire host project with `dotnet run`, it will run every project with the `--no-launch-profile` option. This is because if you run the project with `launch_profile` received, you will have conflicting ports because Aspire uses the port from the profile in the proxy. So, how to work with these properties correctly?

Reply:
The invocation via `dotnet run` does not really use this protocol today, so it is not applicable.

For Visual Studio, `dotnet watch` (work pending), and presumably Rider, the property reflects the launch profile the user requested in the app model. It is the launch profile for the service program, which is distinct from the launch profile for the application host program. It is up to the IDE/CLI to apply the requested launch profile to the service program, and it is up to the developer to ensure that application host launch profile and the service(s) launch profile(s) do not conflict.

Not sure if I fully understand the question; hope this helps anyway.

Reply:
Yes, thank you!